src/anyio/_backends/_asyncio.py

```diff
@@ -2683,13 +2684,24 @@ async def wait_socket_readable(cls, sock: socket.socket) -> None:
             raise BusyResourceError("reading from") from None
 
         loop = get_running_loop()
+        if (
+            sys.platform == "win32"
+            and asyncio.get_event_loop_policy().__class__.__name__
+            == "WindowsProactorEventLoopPolicy"
+        ):
+            selector = _get_selector_windows(loop)
+            add_reader = selector.add_reader
+            remove_reader = selector.remove_reader
+        else:
+            add_reader = loop.add_reader
+            remove_reader = loop.remove_reader
         event = read_events[sock] = asyncio.Event()
-        loop.add_reader(sock, event.set)
+        add_reader(sock, event.set)
```

Comment:
perhaps something like:

```suggestion
        try:
            add_reader(sock, event.set)
        except NotImplementedError:
            # Proactor on Windows does not yet implement add/remove reader
            selector = _get_selector_windows(loop)
            selector.add_reader(sock, event.set)
            remove_reader = selector.remove_reader
        else:
            remove_reader = loop.remove_reader
```